Neos.SiteKickstarter/Resources/Private/Generator/Fusion/Root.fusion

```diff
@@ -1,37 +1,39 @@
 /**
```

Comment:
I think this file should be removed entirely and the default fusion should be adjusted to support document type based rendering.

```
root {
    ... 
    document {
		@position = 'end 9998'
		condition = Neos.Fusion:PrototypeExists {
		    prototypeName = ${q(node).property('_nodeType.name')}
		}
		type = ${q(documentNode).nodeType.name}
	}
	
	default {
		@position = 'end 9999'
		condition = TRUE
		renderPath = '/page'
	}
	...
}
```

Offcourse the `Neos.Fusion:PrototypeExists` object does not exist yet and this would also lead to conflicts with the default expectation there is a template for every  document type.

Reply:
The breakiness could be reduced if we add extend the check to wether the prototyoe that is named after the document extends page 

```
root {
    ... 
    document {
        @position = 'end 9998'
        condition = Neos.Fusion:VerifyPrototype {
            name = ${q(node).property('_nodeType.name')}
            basedOn = `Neos.Neos:Page`
        }
        type = ${q(documentNode).nodeType.name}
    }
```